### ISSUE TITLE
completed-docs should not require documentation on the implementation of function overloads

### ISSUE THREAD

**Issue by UselessPickles** [palantir/tslint#4573]
### Bug Report

- __TSLint version__: 5.13.1
- __TypeScript version__: 3.3.3333
- __Running TSLint via__:  CLI

#### TypeScript code being linted

```ts
/**
 * Documentation    
 * @param x - A parameter
 * @returns A value
 */
export function foo(x: number): number;
/**
 * Documentation
 * @param x - A parameter
 * @returns A value
 */
export function foo(x: string): string;
export function foo(x: number | string): number | string {
    return x;
}
```

with `tslint.json` configuration (excerpt of relevant rule config):

```json
{
    "rules": {
        "completed-docs": [
            true, {
                "functions": {
                    "locations": ["all"]
                }
            }
        ]
    }
}
```

#### Actual behavior

completed-docs fails on the signature of the implementation of overloaded function `foo`.

![Screen Shot 2019-03-11 at 3 41 26 PM](https://user-images.githubusercontent.com/22800095/54153094-1ba04100-4415-11e9-9293-0c15c7551d9f.png)

#### Expected behavior

When a function is overloaded, the signature of the implementation is not a callable signature. It's simply a common signature compatible with all overload signatures so that a single implementation can be written that works with all overload signatures. It is not very useful/necessary to have documentation on the overload implementation, because all usages of the function in code will resolve to one of the overload signatures, and IDEs will present the documentation of the appropriately matched overload signature (and/or will allow the user to toggle between all overload signatures and their documentation). The signature of the implementation itself is never useful to present to the user, because it is not callable.

When a function has multiple signatures (is overloaded), the signature of the implementation itself should not require documentation.

Somewhat related to #4416, yet still probably independent of it.

**Comment by JoshuaKGoldberg** [comment]
Agreed. I think once #4563 goes in, TSLint should consider these to be overloads under the same option.

**Comment by UselessPickles** [comment]
Note that this would similarly be applicable to methods of classes with overload signatures.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
